src/layers/core/solid-polygon-layer/polygon-tesselator-extruded.js

```diff
@@ -117,15 +118,15 @@ function calculateIndices({groupedVertices, wireframe = false}) {
 // * each top vertex is on 3 surfaces
 // * each bottom vertex is on 2 surfaces
 function calculatePositionsJS({groupedVertices, wireframe = false}) {
-  const positions = groupedVertices.map(complexPolygon =>
-    complexPolygon.map(vertices => {
-      const topVertices = [].concat(vertices);
+  const positions = groupedVertices.map(
+    vertices => {
+      const topVertices = Array.prototype.concat.apply([], vertices);
```

Comment:
Yeah, the bug is that applying double map() to groupedVertices changes the sequence of the vertices. So I reverted back to a single map(). 

However, using single map(), I need to use concat.apply() to have topVertices in flattened format for the baseVertices to be correctly calculated.

It took me some time to figure out that
[].concat([[1, 2, 3]]) is [[1, 2, 3]],
Array.prototype.concat.apply([], [[1, 2, 3]]) is [1, 2, 3]

Learned a Javascript lesson....